packages/blade/src/components/Button/BaseButton/__tests__/BaseButton.native.test.tsx

```diff
@@ -0,0 +1,102 @@
+  });
+  it('should throw an error when there is no icon or text passed to the button', () => {
+    try {
+      // @ts-expect-error
+      renderWithTheme(<BaseButton />);
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        expect(error.message).toEqual(
+          `[Blade: BaseButton]: Cannot render a BaseButton without an icon or text`,
+        );
+      }
+    }
+  });
+  it('should render large size button', () => {
+    const buttonText = 'Pay Now';
+    const { toJSON } = renderWithTheme(<BaseButton size="large">{buttonText}</BaseButton>);
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render medium size button', () => {
+    const buttonText = 'Pay Now';
+    const { toJSON } = renderWithTheme(<BaseButton size="medium">{buttonText}</BaseButton>);
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render small size button', () => {
+    const buttonText = 'Pay Now';
+    const { toJSON } = renderWithTheme(<BaseButton size="small">{buttonText}</BaseButton>);
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render xsmall size button', () => {
+    const buttonText = 'Pay Now';
+    const { toJSON } = renderWithTheme(<BaseButton size="xsmall">{buttonText}</BaseButton>);
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render button with icon without text', () => {
+    const { toJSON } = renderWithTheme(<BaseButton icon={CreditCardIcon} />);
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render button with icon with default iconPosition', () => {
+    const buttonText = 'Pay Now';
+    const { toJSON } = renderWithTheme(<BaseButton icon={CreditCardIcon}>{buttonText}</BaseButton>);
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render button with icon with left iconPosition', () => {
+    const buttonText = 'Pay Now';
+    const { toJSON } = renderWithTheme(
+      <BaseButton iconPosition="left" icon={CreditCardIcon}>
+        {buttonText}
+      </BaseButton>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render button with icon with right iconPosition', () => {
+    const buttonText = 'Pay Now';
+    const { toJSON } = renderWithTheme(
+      <BaseButton iconPosition="right" icon={CreditCardIcon}>
+        {buttonText}
+      </BaseButton>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render button with full width', () => {
+    const buttonText = 'Pay Now';
+    const { toJSON } = renderWithTheme(<BaseButton isFullWidth={true}>{buttonText}</BaseButton>);
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render disabled button', () => {
+    const buttonText = 'Pay Now';
+    const onClick = jest.fn();
+    const { toJSON, getByText } = renderWithTheme(
+      <BaseButton isDisabled={true} onClick={onClick}>
+        {buttonText}
+      </BaseButton>,
+    );
+    const button = getByText(buttonText);
+    fireEvent.press(button);
```

Comment:
can we simulate a test to check the styles when pressed? Basically tests which can test different states(focus, active, etc) of button

Reply:
Yeah, tried doing this. Its a bit weird on native. We have to pass `testOnly_pressed={true}` for us to simulate a `pressed` behaviour 🤦 

This means passing `testOnly_pressed` from the root `BaseButton` leading to the same prop being exposed to consumers as well as web which doesn't seem like a good enough tradeoff